src/sql/workbench/contrib/notebook/test/browser/dataResourceDataProvider.test.ts

```diff
@@ -0,0 +1,129 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as assert from 'assert';
+import * as TypeMoq from 'typemoq';
+import * as azdata from 'azdata';
+import * as os from 'os';
+import * as path from 'path';
+import * as fs from 'fs-extra';
+import * as uuid from 'uuid';
+import * as sinon from 'sinon';
+import { DataResourceDataProvider } from '../../browser/outputs/gridOutput.component';
+import { IDataResource } from 'sql/workbench/services/notebook/browser/sql/sqlSessionManager';
+import { ResultSetSummary } from 'sql/workbench/services/query/common/query';
+import { TestNotificationService } from 'vs/platform/notification/test/common/testNotificationService';
+import { TestFileDialogService, TestEditorService } from 'vs/workbench/test/browser/workbenchTestServices';
+import { SerializationService } from 'sql/platform/serialization/common/serializationService';
+import { SaveFormat, ResultSerializer } from 'sql/workbench/services/query/common/resultSerializer';
+import { InstantiationService } from 'vs/platform/instantiation/common/instantiationService';
+import { URI } from 'vs/base/common/uri';
+
+export class TestSerializationProvider implements azdata.SerializationProvider {
+	providerId: string;
+	constructor(providerId: string = 'providerId') { }
+
+	// Write data to file
+	startSerialization(requestParams: azdata.SerializeDataStartRequestParams): Thenable<azdata.SerializeDataResult> {
+		let data: string = '';
+		requestParams.rows.forEach((row) => {
+			row.forEach((element) => {
+				data += element.displayValue + ' ';
+			});
+			data += '\n';
+		});
+		return fs.promises.writeFile(requestParams.filePath, data).then(() => {
```

Comment:
nit: But using async here would make this cleaner. 

```
await fs.promises.writeFile(...);
return { succeeded: true, messages: undefined };
```

Reply:
The startSerialization method in azdata.SerializationProvider is not async so I wasn't able to use await here.

Reply:
async is just syntactic sugar - you can add it to any function that returns a promise.

To do that you'll need to update the return type to be a Promise. But that's fine since a Promise is a Thenable so still fulfills the implementation.